Add button styling and story

rasmuswoelk opened the pull request:
This is my first pull request to stripes-components - feel free to mention if I should be doing anything differently.

You won't see a lot of commits because I initially just cloned the original repo instead of forking it so I just copied my files after forking.

**Changes**
- Added story to button (I have experimented with some additional sub folders to keep the component folder as clean as possible and to make the documentation more manageable)
- Updated some button styling (focus, hover and active styling etc.)
- Added two new styles (warning and success)
- Added mixin for easily changing or adding button styles
- Added two new post css plugins (css-rucksack and postcss-mixins). Important: These needs to be added to stripes-core.

Let me know if you need any more info :)

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/folio-org/stripes-components?pullRequest=99) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/folio-org/stripes-components?pullRequest=99) before we can accept your contribution.<br/><hr/>**Rasmus Wølk** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/folio-org/stripes-components?pullRequest=99) it.</sub>

id-jenkins commented:
yarn run v1.3.2
$ eslint lib && stylelint "lib/**/*.css"

/home/jenkins/workspace/folio-org/stripes-components/PR-99/lib/Button/Button.stories.js
   1:8   warning  'React' is defined but never used             no-unused-vars
   3:22  warning  'withDocs' is defined but never used          no-unused-vars
   8:1   error    Expected indentation of 2 spaces but found 4  indent
   9:1   error    Expected indentation of 2 spaces but found 4  indent
  10:1   error    Expected indentation of 2 spaces but found 4  indent

/home/jenkins/workspace/folio-org/stripes-components/PR-99/lib/Button/readme/index.js
  2:1   error  Expected indentation of 2 spaces but found 4  indent
  2:14  error  Unexpected require()                          global-require
  3:1   error  Expected indentation of 2 spaces but found 4  indent
  3:13  error  Unexpected require()                          global-require
  4:1   error  Expected indentation of 2 spaces but found 4  indent
  4:12  error  Unexpected require()                          global-require

/home/jenkins/workspace/folio-org/stripes-components/PR-99/lib/Button/stories/general.js
   5:1   error  Expected indentation of 2 spaces but found 4            indent
   5:5   error  Expected indentation of 2 space characters but found 4  react/jsx-indent
   5:17  error  A space is required after '{'                           object-curly-spacing
   5:33  error  A space is required before '}'                          object-curly-spacing
   6:1   error  Expected indentation of 4 spaces but found 8            indent
   6:9   error  Expected indentation of 6 space characters but found 8  react/jsx-indent
   7:1   error  Expected indentation of 4 spaces but found 8            indent
   7:9   error  Expected indentation of 6 space characters but found 8  react/jsx-indent
   8:1   error  Expected indentation of 4 spaces but found 8            indent
   8:9   error  Expected indentation of 6 space characters but found 8  react/jsx-indent
   9:1   error  Expected indentation of 4 spaces but found 8            indent
   9:9   error  Expected indentation of 6 space characters but found 8  react/jsx-indent
  10:1   error  Expected indentation of 2 spaces but found 4            indent

/home/jenkins/workspace/folio-org/stripes-components/PR-99/lib/Button/stories/index.js
   1:8  warning  'React' is defined but never used             no-unused-vars
   8:1  error    Expected indentation of 2 spaces but found 4  indent
   9:1  error    Expected indentation of 2 spaces but found 4  indent
  10:1  error    Expected indentation of 2 spaces but found 4  indent

/home/jenkins/workspace/folio-org/stripes-components/PR-99/lib/Button/stories/sizes.js
  4:1  error  Expected indentation of 2 spaces but found 4            indent
  4:5  error  Expected indentation of 2 space characters but found 4  react/jsx-indent
  6:1  error  Expected indentation of 2 spaces but found 4            indent

/home/jenkins/workspace/folio-org/stripes-components/PR-99/lib/Button/stories/styles.js
   5:1   error  Expected indentation of 2 spaces but found 4             indent
   5:5   error  Expected indentation of 2 space characters but found 4   react/jsx-indent
   5:17  error  A space is required after '{'                            object-curly-spacing
   5:39  error  Unnecessarily quoted property 'maxWidth' found           quote-props
   5:58  error  A space is required before '}'                           object-curly-spacing
   6:1   error  Expected indentation of 4 spaces but found 8             indent
   6:9   error  Expected indentation of 6 space characters but found 8   react/jsx-indent
   7:1   error  Expected indentation of 4 spaces but found 12            indent
   7:13  error  Expected indentation of 6 space characters but found 12  react/jsx-indent
   8:1   error  Expected indentation of 4 spaces but found 12            indent
   8:13  error  Expected indentation of 6 space characters but found 12  react/jsx-indent
   9:1   error  Expected indentation of 4 spaces but found 12            indent
   9:13  error  Expected indentation of 6 space characters but found 12  react/jsx-indent
  10:1   error  Expected indentation of 4 spaces but found 12            indent
  10:13  error  Expected indentation of 6 space characters but found 12  react/jsx-indent
  11:1   error  Expected indentation of 4 spaces but found 12            indent
  11:13  error  Expected indentation of 6 space characters but found 12  react/jsx-indent
  12:1   error  Expected indentation of 4 spaces but found 12            indent
  12:13  error  Expected indentation of 6 space characters but found 12  react/jsx-indent
  13:1   error  Expected indentation of 4 spaces but found 8             indent
  13:9   error  Expected indentation of 6 space characters but found 8   react/jsx-indent
  13:12  error  A space is required before closing bracket               react/jsx-tag-spacing
  14:1   error  Expected indentation of 4 spaces but found 8             indent
  14:9   error  Expected indentation of 6 space characters but found 8   react/jsx-indent
  15:1   error  Expected indentation of 4 spaces but found 12            indent
  15:13  error  Expected indentation of 6 space characters but found 12  react/jsx-indent
  16:1   error  Expected indentation of 4 spaces but found 12            indent
  16:13  error  Expected indentation of 6 space characters but found 12  react/jsx-indent
  17:1   error  Expected indentation of 4 spaces but found 12            indent
  17:13  error  Expected indentation of 6 space characters but found 12  react/jsx-indent
  18:1   error  Expected indentation of 4 spaces but found 12            indent
  18:13  error  Expected indentation of 6 space characters but found 12  react/jsx-indent
  19:1   error  Expected indentation of 4 spaces but found 12            indent
  19:13  error  Expected indentation of 6 space characters but found 12  react/jsx-indent
  20:1   error  Expected indentation of 4 spaces but found 8             indent
  20:9   error  Expected indentation of 6 space characters but found 8   react/jsx-indent
  20:12  error  A space is required before closing bracket               react/jsx-tag-spacing
  21:1   error  Expected indentation of 4 spaces but found 8             indent
  21:9   error  Expected indentation of 6 space characters but found 8   react/jsx-indent
  22:1   error  Expected indentation of 4 spaces but found 12            indent
  22:13  error  Expected indentation of 6 space characters but found 12  react/jsx-indent
  23:1   error  Expected indentation of 4 spaces but found 12            indent
  23:13  error  Expected indentation of 6 space characters but found 12  react/jsx-indent
  24:1   error  Expected indentation of 4 spaces but found 12            indent
  24:13  error  Expected indentation of 6 space characters but found 12  react/jsx-indent
  25:1   error  Expected indentation of 2 spaces but found 4             indent

✖ 77 problems (74 errors, 3 warnings)
  71 errors, 0 warnings potentially fixable with the `--fix` option.

info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

rasmuswoelk commented:
Updated the pull request with the following:

- Removed css-rucksack and postcss-mixins after was made aware that we need to follow CSS4 features as much as possible
- Fixed all linting bugs

cherewaty commented:
@rasmuswoelk you should click that link to sign the CLA, then it won't bug you about that on future PRs.

cherewaty commented:
https://github.com/samhaengcom/stripes-components/blob/f793d86fe9660a7d70d40f6011ad5ed128fa1015/lib/Button/Button.css#L432 expands the height of the buttons at smaller sizes, which can cause situations like this:

<img width="426" alt="screen shot 2017-11-23 at 10 11 28 am" src="https://user-images.githubusercontent.com/230597/33181682-c34880f6-d036-11e7-8594-f421623c5707.png">

Is that the behavior we want?

rasmuswoelk commented:
@cherewaty 

It's probably not the exact behaviour we want however clickable elements should be at least 42x42px on mobile. The design will probably also change on mobile once we get to it. For now we could remove that breakpoint.

Note: The breakpoint and the change of height was there when I started looking at the component (as well as a lot of other classes and functionality) so I didn't wan't to remove it. We should probably re-visit the styling and the props of the component, remove unnecessary styling/logic and then fix it wherever the changes might break something. E.g. The component has two props for removing margin: marginBottom0 and bottomMargin0.

Regarding the CLA: I have clicked on the link and agreed several times. I am not sure why it won't update or save.

JohnC-80 commented on lib/Button/Button.css:
```diff
@@ -44,41 +47,66 @@ a.button {
     border: 1px solid var(--primary);
     color: #fff;
 
-    &:hover {
```
Removing the hover styles - any particular reason? Intended? Just curious.
